source/slang/slang-lower-to-ir.cpp

```diff
@@ -10636,6 +10644,30 @@ struct DeclLoweringVisitor : DeclVisitor<DeclLoweringVisitor, LoweredValInfo>
         FunctionDeclBase* decl,
         bool emitBody = true)
     {
+        bool isFromDifferentModule = isDeclInDifferentModule(context, decl);
+        if (isFromDifferentModule && isForceInlineEarly(decl))
```

Comment:
It's a minor point, but this could conceivably be extended to apply to any imported declaration, and not just function-like declarations. That said, I'm not sure why any other kind of symbol would ever need to support pre-linking, so it's probably just fine to keep the logic here.